packages/tpetra/core/src/Tpetra_Details_StaticView.cpp

```diff
@@ -123,6 +123,33 @@ void finalize_hip_host_pinned_memory ()
 }
 #endif // KOKKOS_ENABLE_HIP
 
+#ifdef KOKKOS_ENABLE_SYCL
```

Comment:
Tpetra has different implementations for different architectures.  Should these implementations actually be in Kokkos rather than Tpetra?

Reply:
I was just repeating what was done above for `Cuda` and `HIP`. If you want this to be refactored, it should be done in a separate pull request IMHO.

Reply:
technically you probably want to refactor all this in a way that you just support Kokkos::DefaultExecutionSpace and Kokkos::DefaultHostExecutionSpace (we talked about that in the Tpetra meeting). This is not something which should be in Kokkos, I mean what you create here are Tpetra data structures not Kokkos's. You could template them or as I said just create them for the default execution spaces, which then are the correct ones based on the Kokkos configurations you build with.